draft-ietf-tls-tls13.md

```diff
@@ -3025,17 +2937,263 @@ processing the server's Certificate, CertificateVerify, and
+in the 0-RTT case are the messages from the handshake where the
+ServerConfiguration was established.
+
+Note 2: The Handshake Context for the last three rows does not include
+any 0-RTT handshake messages, regardless of whether 0-RTT is used.
+
+
+
+####  Certificate
+
+When this message will be sent:
+
+> The server MUST send a Certificate message whenever the agreed-upon
+key exchange method uses certificates for authentication (this
+includes all key exchange methods defined in this document except PSK).
+
+> The client MUST send a Certificate message whenever the server has
+requested client authentication via a CertificateRequest message
+({{certificate-request}}) or when the EarlyDataIndication indicates
+client authentication. This message is only sent if the server
+requests a certificate. If no suitable certificate is available, the
+client MUST send a Certificate message containing no certificates.
+
+Meaning of this message:
+
+> This message conveys the endpoint's certificate chain to the peer.
+
+> The certificate MUST be appropriate for the negotiated cipher suite's key
+exchange algorithm and any negotiated extensions.
+
+Structure of this message:
+
+%%% Authentication Messages
+       opaque ASN1Cert<1..2^24-1>;
+
+       struct {
+            opaque certificate_request_context<0..255>;
+           ASN1Cert certificate_list<0..2^24-1>;
+       } Certificate;
+
+certificate_request_context:
+: If this message is in response to a CertificateRequest, the
+  value if certificate_request_context in that message. Otherwise,
+  in the case of server authentication or client authentication
+  in 0-RTT, this field SHALL be zero length.
+
+certificate_list
+: This is a sequence (chain) of certificates. The sender's
+  certificate MUST come first in the list. Each following
+  certificate SHOULD directly certify one preceding it. Because
+  certificate validation requires that trust anchors be distributed
+  independently, a certificate that specifies a
+  trust anchor MAY be omitted from the chain, provided that
+  supported peers are known to possess any omitted certificates.
+{:br }
+
+Note: Prior to TLS 1.3, "certificate_list" ordering required each certificate
+to certify the one immediately preceding it,
+however some implementations allowed some flexibility. Servers sometimes send
+both a current and deprecated intermediate for transitional purposes, and others
+are simply configured incorrectly, but these cases can nonetheless be validated
+properly. For maximum compatibility, all implementations SHOULD be prepared to
+handle potentially extraneous certificates and arbitrary orderings from any TLS
+version, with the exception of the end-entity certificate which MUST be first.
+
+The server's certificate list MUST always be non-empty. A client MAY
+MAY send no certificates if it does not have an appropriate
+certificate to send in response to the server's authentication
+request.
```

Comment:
I suggest changing the "MAY send no certificates" to "MAY send an empty certificate list" to stress the point that clients MUST send an empty list when a cert is requested but it cannot comply, rather than not sending the message at all.